### PR TITLE
Introduce kludge to make "isEnabledForCluster" work again for kube object menu items PART 4

### PR DESCRIPTION
**Pull request by jansav**
Part 4 of #5800 

Signed-off-by: Janne Savolainen <janne.savolainen@live.fi>

### CODE REVIEW

File: src/renderer/components/kube-object-menu/kube-object-menu-item-injection-token.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import type { IComputedValue } from "mobx";
+import type { KubeObjectMenuProps } from "./kube-object-menu";
+
+export interface KubeObjectMenuItem {
+  kind: string;
+  apiVersions: string[];
+  enabled: IComputedValue<boolean>;
+
+  // TODO: Figure out why "KubeObject" does not work here.
+  Component: React.ElementType<KubeObjectMenuProps<any>>;
```
Comment (Nokel81) [inline src/renderer/components/kube-object-menu/kube-object-menu-item-injection-token.ts]: The reason that using `KubeObject` here doesn't work is that the components being registered say that they require `class T extends KubeObject` which is not the same and typescript rightly complains that any arbitrary instance unrelated to `T` could be passed in.

Personally, I would use `KubeObject` here (and factor out this type as:

```ts
export type KubeObjectMenuItemComponent = React.ElementType<KubeObjectMenuProps<KubeObject>>;
```

and then at the uses of it I would add an `as KubeObjectMenuItemComponent` cast.

File: src/renderer/components/kube-object-menu/kube-object-menu-items.injectable.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        ),
+
+        sortBy((item) => item.orderNumber),
+        map((item) => item.Component),
+      ),
+    );
+  },
+
+  lifecycle: lifecycleEnum.keyedSingleton({
+    getInstanceKey: (di, { kubeObject }: { kubeObject: KubeObject }) =>
```
Comment (Nokel81) [inline src/renderer/components/kube-object-menu/kube-object-menu-items.injectable.ts]: shouldn't it be `KubeObject | undefined`?

**Review by Nokel81 — APPROVED** [review]

File: src/renderer/components/kube-object-menu/kube-object-menu-item-injection-token.ts (hunk 1 above)
Comment (jansav) [reply src/renderer/components/kube-object-menu/kube-object-menu-item-injection-token.ts]: Changed.

File: src/renderer/components/kube-object-menu/kube-object-menu-items.injectable.ts (hunk 2 above)
Comment (jansav) [reply src/renderer/components/kube-object-menu/kube-object-menu-items.injectable.ts]: Hmm. In here it looks like it yeah, but from the perspective of usage (kube-object-menu) it doesn't. I'd would like to be sure it's not undefined.

**Pull request merged**